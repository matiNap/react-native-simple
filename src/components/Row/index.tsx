import React from 'react';
import { StyleSheet, View } from 'react-native';
import { getThemeColor, getProperty } from '../../helpers';
import useTheme from '../ThemeProvider/useTheme';
import Props from '../../types/Row';

export default function ({
    children,
    style,
    backgroundColor,
    width,
    justifyContent,
    align,
}: Props) {
    const theme = useTheme();
    const { Row } = theme;
    return (
        <View
            style={StyleSheet.flatten([
                getProperty([Row?.style]),
                styles.container,
                {
                    width: getProperty([Row?.width, width]),
                    justifyContent: getProperty([
                        Row?.justifyContent,
                        justifyContent,
                    ]),
                    backgroundColor: getThemeColor(
                        getProperty([Row?.backgroundColor, backgroundColor]),
                        theme.currentPalette,
                    ),

                    alignSelf: getProperty([Row?.align, align]),
                },
                style,
            ])}
        >
            {children}
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        width: '100%',
        flexDirection: 'row',
    },
});
