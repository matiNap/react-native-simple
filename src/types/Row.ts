import { StyleProp, ViewStyle } from 'react-native';
import { ReactNode } from 'react';
import { SimpleBackgroundColor } from '.';

export interface LocalRowProps {
    style?: StyleProp<ViewStyle>;
    backgroundColor?: SimpleBackgroundColor;
    width?: string | number;
    justifyContent?:
        | 'space-between'
        | 'flex-start'
        | 'flex-end'
        | 'center'
        | 'space-around'
        | 'space-evenly'
        | undefined;
    align?:
        | 'flex-start'
        | 'flex-end'
        | 'center'
        | 'auto'
        | 'stretch'
        | 'baseline'
        | undefined;
}

export default interface Props extends LocalRowProps {
    children: ReactNode[] | ReactNode;
}
