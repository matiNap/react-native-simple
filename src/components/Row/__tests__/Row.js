import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import Row from '../index';
import toJson from 'enzyme-to-json';

describe('Row component', () => {
    it('Should render', () => {
        const wrapper = shallow(<Row />);

        expect(wrapper.length).toBe(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render with style', () => {
        const wrapper = shallow(
            <Row
                style={{ backgroundColor: 'red' }}
                backgroundColor="blue"
                align="center"
            />,
        );

        expect(wrapper.length).toBe(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render with childs', () => {
        const wrapper = shallow(
            <Row>
                <View />
                <View />
                <View />
                <View />
            </Row>,
        );

        expect(wrapper.length).toBe(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
