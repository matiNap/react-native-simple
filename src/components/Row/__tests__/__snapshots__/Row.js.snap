// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component Should render 1`] = `
<View
  style={
    Object {
      "alignSelf": null,
      "backgroundColor": null,
      "flexDirection": "row",
      "justifyContent": null,
      "width": null,
    }
  }
/>
`;

exports[`Row component Should render with childs 1`] = `
<View
  style={
    Object {
      "alignSelf": null,
      "backgroundColor": null,
      "flexDirection": "row",
      "justifyContent": null,
      "width": null,
    }
  }
>
  <View />
  <View />
  <View />
  <View />
</View>
`;

exports[`Row component Should render with style 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "red",
      "flexDirection": "row",
      "justifyContent": null,
      "width": null,
    }
  }
/>
`;
